src/pinnwand/handler/api_curl.py

```diff
@@ -14,12 +15,18 @@ def check_xsrf_cookie(self) -> None:
     def get(self) -> None:
         raise tornado.web.HTTPError(400)
 
-    def post(self) -> None:
-        if defensive.ratelimit(self.request, area="create"):
+    def write_error(self, status_code: int, **kwargs: Any) -> None:
+        type_, exc, traceback = kwargs["exc_info"]
+
+        if type_ == error.ValidationError:
+            self.set_status(400)
+            self.write(str(exc))
+        elif type_ == error.RatelimitError:
             self.set_status(429)
             self.write("Enhance your calm, you have exceeded the ratelimit.")
```

Comment:
I think we might need to defer back to the default error writer here, to avoid silently ignoring other errors.
```suggestion
        elif type_ == error.RatelimitError:
            self.set_status(429)
            self.write("Enhance your calm, you have exceeded the ratelimit.")
        else:
            super().write_error(status_code, **kwargs)
```

Reply:
7008f979c8b3f62caf91485d1426c631ed1376f2